ducktape/services/service.py

```diff
@@ -100,11 +102,29 @@ def start(self):
 
         self.logger.debug(self.who_am_i() + ": killing processes and attempting to clean up before starting")
         for node in self.nodes:
-            # Added precaution - kill running processes
-            self._kill_running_processes(node)
-            self.force_clean_node(node)
-            self.stop_node(node)
-            self.clean_node(node)
+            # Added precaution - kill running processes, clean persistent files
+            # try/except for each step, since each of these steps may fail if there are no processes
+            # to kill or no files to remove
+
+            try:
+                self._kill_running_processes(node)
+            except:
+                pass
+
+            try:
+                self.force_clean_node(node)
+            except:
+                pass
+
+            try:
+                self.stop_node(node)
```

Comment:
On "nuclear cleanup", agreed, you need some info about the cluster type. I'm a bit worried about relying on that option though -- anything sufficiently complex is going to turn into a mess. For example, in system tests for clients which use a wide variety of languages, is there really going to be a reasonable nuclear option? Or if there is, is it going to require a ton of steps to cover all the possible process types?

For file cleanup, I like that idea though we need to figure out where global-ish configs like that should come from.

For process cleanup, agreed a more uniform approach to processes would help. One option would be for the pid files to be stored in a standard location so that its easy to just scan through and kill -9 them if they still exist. Of course that's another global-ish config that needs to be specified on a per-project basis.